sceptre/config/graph.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+
+"""
+sceptre.config.graph
+This module implements a StackConfig graph, which stores a directed graph
+of a stack's dependencies.
+"""
+
+import logging
+import networkx as nx
+from sceptre.exceptions import CircularDependenciesError
+
+
+class StackDependencyGraph(object):
+    """
+    A Directed Graph representing the relationship between Stack config
+    dependencies. Responsible for initalizing the graph object based on
+    a given inital stack config path.
+
+    :param inital_config_path: The inital stack config you want to generate
+    the graph from
+    :type inital_config_path: str
+    """
+
+    def __init__(self, dependency_map=None):
+        self.logger = logging.getLogger(__name__)
+        self.graph = nx.DiGraph()
+        if dependency_map is None:
+            dependency_map = {}
```

Comment:
Why not set `dependency_map={}` in the function definition rather than `dependency_map=None`? Then these two lines wouldn't be needed.

Reply:
This is the whole https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments thingy

Reply:
Aha, makes sense.